lib/bcdice/game_system/Garactier.rb

```diff
@@ -0,0 +1,479 @@
+        unless m
+          return nil
+        end
+
+        modifier = Arithmetic.eval(m[1] || "", RoundType::FLOOR) || 0
+
+        target_flag = !m[2].nil?
+        dice_result = roll_dice_with_modifier(modifier)
+
+        if target_flag
+          target = Arithmetic.eval(m[3] || "", RoundType::FLOOR) || 0
+          r = determine_target_result("", dice_result[:total], target, dice_result[:critical], dice_result[:fumble])
+        else
+          r = determine_no_target_result("", dice_result[:total], dice_result[:critical], dice_result[:fumble])
+        end
+        r.text = "(#{m[0]}) ＞ #{dice_result[:dice_sum]}[#{dice_result[:dice_list].join(',')}]#{m[1]} ＞ #{dice_result[:total]} ＞ #{r.text}"
+        return r
+      end
+
+      # 基準値(modifier)をもとに2d6+基準値の判定を行う
+      def roll_dice_with_modifier(modifier)
+        dice_list = randomizer.roll_barabara(2, 6)
+        dice_sum = dice_list.sum
+        total = dice_sum + modifier
+        critical_flag = dice_list.count(6) == 2
+        fumble_flag = dice_list.count(1) == 2
+        return {dice_list: dice_list, dice_sum: dice_sum, total: total, critical: critical_flag, fumble: fumble_flag}
+      end
+
+      # roll_typeごとにResultを作成（目標値あり）
+      def determine_target_result(roll_type, total, target, critical, fumble)
+        case roll_type
+        when "H"
+          if critical
+            Result.critical("クリティカル命中")
+          elsif fumble
+            Result.fumble("ファンブル")
+          elsif total >= target + 4
+            Result.success("急所命中")
+          elsif total >= target
+            Result.success("命中")
+          else
+            Result.failure("失敗")
+          end
+        when "D"
+          if critical
+            Result.critical("クリティカル")
+          elsif fumble
+            Result.fumble("ファンブル")
+          elsif total >= target
+            Result.success("回避成功")
+          elsif total >= target - 4
+            Result.failure("半減命中")
+          else
+            Result.failure("失敗")
+          end
+        when "M"
+          # 抵抗判定は基準値以上(激情)のほうが悪い効果のことが多いためResultを反転[6,6]の場合にファンブル
+          if critical
+            Result.fumble("必ず激情")
+          elsif fumble
+            Result.critical("必ず平静")
+          elsif total >= target
+            Result.failure("激情")
+          else
+            Result.success("平静")
+          end
+        else
+          if critical
+            Result.critical("クリティカル")
+          elsif fumble
+            Result.fumble("ファンブル")
+          elsif total >= target
+            Result.success("成功")
+          else
+            Result.failure("失敗")
+          end
+        end
+      end
+
+      # roll_typeごとにResultを作成（目標値なし）
+      def determine_no_target_result(roll_type, total, critical, fumble)
+        case roll_type
+        when "S"
+          if critical
+            Result.critical("クリティカル")
+          elsif fumble
+            Result.fumble("ファンブル")
+          else
+            Integer success_level = (total - 4) / 2
```

Comment:
```suggestion
            success_level = (total - 4) / 2
```

Rubyでは整数と整数の割り算は整数になるため、 `Integer` が切り捨ての処理を意図しているならば不要です。

Reply:
削除、修正行いました。
Ruby不勉強で何度も申し訳ありません。